bridge/slack/handlers.go

```diff
@@ -64,8 +64,12 @@ func (b *Bslack) handleSlackClient(messages chan *config.Message) {
 			b.populateUsers()
 		case *slack.ConnectedEvent:
 			b.si = ev.Info
-			b.populateChannels()
-			b.populateUsers()
+
+			// Populating channels and users can take a while so do it async.
+			go func() {
+				b.populateChannels()
+				b.populateUsers()
+			}()
 		case *slack.InvalidAuthEvent:
```

Comment:
This was sync with a reason, so that we kept blocking here and don't start receiving messsages where don't have the info about the channel or users. Maybe for big channels (because that takes a long time with the new cursor API) we can do ad-hoc lookup of a user instead (or combined with) loading them all at once. 
The populateChannels() shouldn't take long and can be kept sync.

WDYT ?

Reply:
You are quite right and in retrospective I am removing the async stuff entirely. Just limiting this PR to the scope of not running multiple `populateChannels()` and `populateUsers()` in parallel.